utils/encryption.py

```diff
@@ -0,0 +1,100 @@
+import getpass
+import json
+import os
+from base64 import b64encode, b64decode
+from Crypto.Cipher import ChaCha20_Poly1305
+from Crypto.Random import get_random_bytes
+from Crypto.Protocol.KDF import scrypt
+
+from utils.logger import log
+
+
+def encrypt(pt, password):
+    salt = get_random_bytes(32)
+    key = scrypt(password, salt, key_len=32, N=2 ** 20, r=8, p=1)
+    nonce = get_random_bytes(12)
+    cipher = ChaCha20_Poly1305.new(key=key, nonce=nonce)
+    ct, tag = cipher.encrypt_and_digest(pt)
+    json_k = ["nonce", "salt", "ct", "tag"]
+    json_v = [b64encode(x).decode("utf-8") for x in (nonce, salt, ct, tag)]
+    result = json.dumps(dict(zip(json_k, json_v)))
+
+    return result
+
+
+def decrypt(ct, password):
+    try:
+        b64Ct = json.loads(ct)
+        json_k = ["nonce", "salt", "ct", "tag"]
+        json_v = {k: b64decode(b64Ct[k]) for k in json_k}
+
+        key = scrypt(password, json_v["salt"], key_len=32, N=2 ** 20, r=8, p=1)
+        cipher = ChaCha20_Poly1305.new(key=key, nonce=json_v["nonce"])
+        ptData = cipher.decrypt_and_verify(json_v["ct"], json_v["tag"])
+
+        return ptData
+    except (KeyError, ValueError):
+        print("Incorrect Password.")
+        exit(0)
+
+
+def create_encrypted_config(data, file_path):
+    if isinstance(data, dict):
+        data = json.dumps(data)
+    payload = bytes(data, "utf-8")
+    log.info("Create a password for the credential file")
+    cpass = getpass.getpass(prompt="Credential file password: ")
+    vpass = getpass.getpass(prompt="Verify credential file password: ")
+    if cpass == vpass:
+        result = encrypt(payload, cpass)
+        with open(file_path, "w") as f:
+            f.write(result)
+        log.info("Credentials safely stored.")
+    else:
+        print("Password and verify password do not match.")
+        exit(0)
+
+
+def load_encrypted_config(config_path):
+    log.info("Reading credentials from: " + config_path)
+    with open(config_path, "r") as json_file:
+        data = json_file.read()
+    try:
+        if "nonce" in data:
+            password = getpass.getpass(prompt="Password: ")
```

Comment:
This is unclear if this is asking for the password to decrypt the file, or your password for the site. 

Is there a way to enter this password without the prompt or to store it somewhere?

Reply:
Storing the password kind of defeats the purpose of this, we don't want user information left in a state that is exposed or easily retrievable.